crate/server/src/core/implementation.rs

```diff
@@ -242,20 +242,34 @@ impl KMS {
                 )
             })?;
 
+        let private_key_mask = request
+            .private_key_attributes
+            .as_ref()
+            .and_then(|attr| attr.cryptographic_usage_mask);
+
+        let public_key_mask = request
+            .public_key_attributes
+            .as_ref()
+            .and_then(|attr| attr.cryptographic_usage_mask);
+
         // Check that the cryptographic algorithm is specified.
-        let cryptographic_algorithm = &any_attributes.cryptographic_algorithm.ok_or_else(|| {
+        let cryptographic_algorithm = any_attributes.cryptographic_algorithm.ok_or_else(|| {
             KmsError::InvalidRequest(
                 "the cryptographic algorithm must be specified for key pair creation".to_string(),
             )
         })?;
 
-        let key_pair = match &cryptographic_algorithm {
-            CryptographicAlgorithm::ECDH => {
-                let dp = any_attributes
+        let key_pair = match cryptographic_algorithm {
+            // EC, ECDSA and ECDH posses the same FIPS restrictions for curves.
```

Comment:
```suggestion
            // EC, ECDSA and ECDH possess the same FIPS restrictions for curves.
```

Reply:
fixed in commit ab48ee823b87dd914cdb2b71a7609bbc8a552bf9